pkg/startup/startup.go

```diff
@@ -244,6 +222,61 @@ func Run() {
 		log.WithError(err).Errorf("Unable to ensure network for os")
 		terminate()
 	}
+
+	if !runOnce {
+		autoDetectPollingInterval := DEFAULT_AUTODETECT_POLL_INTERVAL
+		if os.Getenv("AUTODETECT_POLL_INTERVAL") != "" {
+			autoDetectPollingInterval, _ = time.ParseDuration(os.Getenv("AUTODETECT_POLL_INTERVAL"))
+		}
+		for {
+			select {
+			case <-time.After(autoDetectPollingInterval):
+				checkConflicts = setupIPAndSubnets(ctx, node, cli)
+				if checkConflicts {
+					checkNodeIPConflictCheck(ctx, node, cli)
+					// Apply the updated node resource.
+					if _, err := CreateOrUpdate(ctx, cli, node); err != nil {
+						log.WithError(err).Errorf("Unable to set node resource configuration")
+						terminate()
+					}
+				}
+			}
+		}
+	}
+}
+
+func setupIPAndSubnets(ctx context.Context, node *api.Node, cli client.Interface) bool {
+	// Configure and verify the node IP addresses and subnets.
+	checkConflicts, err := configureIPsAndSubnets(node)
+	if err != nil {
+		clearv4 := os.Getenv("IP") == "autodetect"
+		clearv6 := os.Getenv("IP6") == "autodetect"
+		if node.ResourceVersion != "" {
+			// If we're auto-detecting an IP on an existing node and hit an error, clear the previous
+			// IP addresses from the node since they are no longer valid.
+			clearNodeIPs(ctx, cli, node, clearv4, clearv6)
+		}
+		terminate()
```

Comment:
We need to be careful about the use of `terminate()` here - I don't think we want to terminate in the daemon version of this code. Rather we should return an error and catch it in a retry loop of some kind.

I guess if we terminate then `runsv` will start the service again from the beginning, so maybe not the end of the world.